tests/console/apache.pm

```diff
@@ -0,0 +1,156 @@
+    assert_script_run 'mkdir -p /srv/www/vhosts/dummy-host.example.com';
+    assert_script_run 'touch /srv/www/vhosts/dummy-host.example.com/listed_test_file';
+
+    # Create separate vhost for 'localhost'
+    assert_script_run 'cp /etc/apache2/vhosts.d/vhost.template /etc/apache2/vhosts.d/localhost.conf';
+    assert_script_run 'sed -i "s/dummy-host.example.com/localhost/g" /etc/apache2/vhosts.d/localhost.conf';
+    assert_script_run 'mkdir -p /srv/www/vhosts/localhost';
+    assert_script_run 'echo "This is the local host" > /srv/www/vhosts/localhost/index.html';
+
+    # Create separate vhost for 'foo.bar'
+    assert_script_run 'cp /etc/apache2/vhosts.d/vhost.template /etc/apache2/vhosts.d/foo.bar.conf';
+    assert_script_run 'sed -i "s/dummy-host.example.com/foo.bar/g" /etc/apache2/vhosts.d/foo.bar.conf';
+    assert_script_run 'mkdir -p /srv/www/vhosts/foo.bar';
+    assert_script_run 'echo "This is the foo host" > /srv/www/vhosts/foo.bar/index.html';
+    assert_script_run 'echo "127.0.0.1 foo.bar" >> /etc/hosts';
+    assert_script_run 'echo "::1 foo.bar" >> /etc/hosts';
+
+    # Check the syntax and reload the server
+    assert_script_run 'apachectl -t';
+    systemctl 'reload apache2';
+
+    # Check the 'localhost' vhost via IPv4 and IPv6
+    assert_script_run 'curl -H "Host: localhost" http://[::1]/ | grep "This is the local host"';
+    assert_script_run 'curl http://localhost/ | grep "This is the local host"';
+
+    # Check the 'foo.bar' vhost via IPv4 and IPv6
+    assert_script_run 'curl -4 http://foo.bar | grep "This is the foo host"';
+    assert_script_run 'curl -6 http://foo.bar/ | grep "This is the foo host"';
+
+    # Works we have also special vhost listening on port 85 both IPv4 and IPv6
+    assert_script_run 'curl http://localhost:85/ | grep "listed_test_file"';
+    assert_script_run 'curl http://[::1]:85/ | grep "listed_test_file"';
+
+    # We stop current webserver and prepare another http2-prefork environment
+    systemctl 'stop apache2';
+    zypper_call 'in libapr1 apache2 apache2-prefork';
+    assert_script_run 'mkdir -p /tmp/prefork';
+    assert_script_run 'sed "s_\(/var/log/apache2\|/var/run\)_/tmp/prefork_; s/80/8080/" /usr/share/doc/packages/apache2/httpd.conf.default > /tmp/prefork/httpd.conf';
+
+    # Run and test this new environment
+    assert_script_run 'httpd2-prefork -f /tmp/prefork/httpd.conf';
+    assert_script_run 'ps aux | grep "\-f /tmp/prefork/httpd.conf" | grep httpd2-prefork';
+
+    # Run and test the old environment too
+    assert_script_run 'rm /var/run/httpd.pid';
+    systemctl 'start apache2';
+    assert_script_run 'ps aux | grep "\-f /etc/apache2/httpd.conf" | grep httpd-prefork';
+
+    # Test both instances
+    assert_script_run 'curl http://localhost:80/';
+    assert_script_run 'curl http://localhost:8080/';
+
+    # Stop both instances
+    assert_script_run 'killall -TERM httpd2-prefork';
```

Comment:
systemctl will not stop it properly ?

Reply:
there is no service for it ... ok

Reply:
There are two instances running at the same time - one is started and then stopped from systemd and the other is started manually and then killed over here...